packages/webview_flutter/lib/webview_flutter.dart

```diff
@@ -79,6 +79,9 @@ typedef void PageStartedCallback(String url);
 /// Signature for when a [WebView] has finished loading a page.
 typedef void PageFinishedCallback(String url);
 
+/// Signature for when a [WebView] has failed to load a resource.
+typedef void ReceivedErrorCallback(String errorCode, String description);
```

Comment:
When reading this my gut was that we should have an error class though I'm not finding good arguments any way 😄 (slight pro of an object is that it's easier for users to pass the error information around within their app)

I wonder if you had a reason to prefer not introducing an error class.

Reply:
I was in favor of a data class as well. I chose not to create one to be consistent with the style of the `WebViewPlatformCallbacksHandler`.  Method
```
void onJavaScriptChannelMessage(String channel, String message);
```
and
```
FutureOr<bool> onNavigationRequest({String url, bool isForMainFrame});
```
don't use them either.

However, since I need more than 2 parameters now, I created a `WebResourceError` class.